api/apps/v1alpha1/nimcache_types.go

```diff
@@ -110,13 +110,17 @@ type GPUSpec struct {
 
 // DataStoreSource references a model stored on NVIDIA DataStore service.
 type DataStoreSource struct {
-	// The endpoint for datastore
+	// HF-compatible datastore endpoint
+	// +kubebuilder:validation:Pattern=`^http(s)?://.*$`
 	Endpoint string `json:"endpoint"`
-	// Name of either model/checkpoint or dataset to download
-	ModelName      *string `json:"modelName,omitempty"`
-	CheckpointName *string `json:"checkpointName,omitempty"`
-	DatasetName    *string `json:"datasetName,omitempty"`
-	// The name of an existing auth secret containing the AUTH_TOKEN"
+	// +kubebuilder:default="default"
+	Namespace string `json:"namespace"`
```

Comment:
Can you make this optional? And also add a comment as to what this field is? At first glance, we might think this is a k8s namespace.

```suggestion
	Namespace string `json:"namespace,omitempty"`
```